File: frontend/src/types/reports.ts
export interface BrandingConfig {
  company_name: string;
  logo_url?: string;
  brand_color: string;
  author: string;
}

export interface ReportExportRequest {
  format: 'PDF' | 'DOCX' | 'PPTX';
  title: string;
  include_executive_summary: boolean;
  include_charts: boolean;
  include_ai_explanations: boolean;
  branding: BrandingConfig;
}

export interface ReportExportResponse {
  id: string;
  repository_id: string;
  format: 'PDF' | 'DOCX' | 'PPTX';
  title: string;
  executive_summary: string;
  branding_info: BrandingConfig;
  file_name: string;
  file_size_bytes: number;
  download_url: string;
  generated_at: string;
}

File: backend/app/schemas/reports.py
import uuid
from datetime import datetime
from typing import Dict, List, Optional, Any
from pydantic import BaseModel, Field

class BrandingConfig(BaseModel):
    company_name: str = "CodeGuardian AI Corp"
    logo_url: Optional[str] = "https://codeguardian.ai/logo.png"
    brand_color: str = "#4f46e5"
    author: str = "Automated Security & Audit Engine"

class ReportExportRequest(BaseModel):
    format: str = Field(default="PDF", description="Export Format: PDF, DOCX, or PPTX")
    title: str = Field(default="Repository Audit & Security Evaluation Report")
    include_executive_summary: bool = True
    include_charts: bool = True
    include_ai_explanations: bool = True
    branding: BrandingConfig = Field(default_factory=BrandingConfig)

class ReportExportResponse(BaseModel):
    id: uuid.UUID
    repository_id: uuid.UUID
    format: str
    title: str
    executive_summary: str
    branding_info: Dict[str, Any]
    file_name: str
    file_size_bytes: int
    download_url: str
    generated_at: datetime

    class Config:
        from_attributes = True
